src/jit/codegenxarch.cpp

```diff
@@ -2631,6 +2489,79 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
 
 #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
 //------------------------------------------------------------------------
+// getReturnRegistersBasedOnTypes: Returns the return registers for a specific struct types.
+//
+// Arguments:
+//    structDesc    - struct classification description.
+//    type0         - the type of the first eightbyte to be returned.
+//    type1         - the type of the second eightbyte to be returned.
+//    retRegPtr0    - returns the register for the first eightbyte.
+//    retRegPtr1    - returns the register for the second eightbyte.
+//
+
+void
+CodeGen::getReturnRegistersBasedOnTypes(const SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR& structDesc,
+                                        var_types type0,
+                                        var_types type1,
+                                        regNumber* retRegPtr0,
+                                        regNumber* retRegPtr1)
+{
+    *retRegPtr0 = REG_NA;
+    *retRegPtr1 = REG_NA;
+
+    bool firstIntUsed = false;
+    bool firstFloatUsed = false;
+
+    if (type0 != TYP_UNKNOWN)
+    {
+        if (structDesc.IsIntegralSlot(0))
```

Comment:
Help me understand why do we need structDesc given var_types of first and second 8-bytes?  Can't we just determine the return register based on type0 and type1 alone?